### PR TITLE
{lib}[GCCcore/11.2.0] UCX-1.11.2

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff --git a/ucx-1.11.2-orig/src/uct/ib/base/ib_device.h b/ucx-1.11.2/src/uct/ib/base/ib_device.h
```
Comment (Micket) [inline easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch:1]: Please include a short description of why the change is here and who the author is (you)

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/u/UCX/UCX-1.11.2-GCCcore-11.2.0.eb [hunk 2]
```diff
@@ -14,10 +14,15 @@ toolchainopts = {'pic': True}
 
 source_urls = ['https://github.com/openucx/ucx/releases/download/v%(version)s']
 sources = ['%(namelower)s-%(version)s.tar.gz']
-patches = ['UCX-1.10.0-dynamic_modules.patch']
+patches = [
+    'UCX-1.10.0-dynamic_modules.patch',
+    'UCX-1.11.2-missing_header.patch',
+
```
Comment (Micket) [inline easybuild/easyconfigs/u/UCX/UCX-1.11.2-GCCcore-11.2.0.eb:20]:
```suggestion
```

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch (hunk 1 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch:3]: Done. Sorry for that and thanks for spotting!

File: easybuild/easyconfigs/u/UCX/UCX-1.11.2-GCCcore-11.2.0.eb (hunk 2 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/u/UCX/UCX-1.11.2-GCCcore-11.2.0.eb]: Done. Thanks for spotting!

File: easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Author: J. Sassmannshausen / GSTT 
+diff --git a/ucx-1.11.2-orig/src/uct/ib/base/ib_device.h b/ucx-1.11.2/src/uct/ib/base/ib_device.h
+index ac69544..5eb39b0 100644
+--- a/ucx-1.11.2-orig/src/uct/ib/base/ib_device.h
++++ b/ucx-1.11.2/src/uct/ib/base/ib_device.h
+@@ -8,6 +8,7 @@
+ #ifndef UCT_IB_DEVICE_H
+ #define UCT_IB_DEVICE_H
+ 
++#include "string.h"
```
Comment (bartoldeman) [inline easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch:11]: you should use `<string.h>` for system headers

File: easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ #include <uct/api/uct.h>
+diff --git a/ucx-1.11.2-orig/src/uct/ib/base/ib_verbs.h b/ucx-1.11.2/src/uct/ib/base/ib_verbs.h
+index 524139c..4822477 100644
+--- a/ucx-1.11.2-orig/src/uct/ib/base/ib_verbs.h
++++ b/ucx-1.11.2/src/uct/ib/base/ib_verbs.h
+@@ -19,6 +19,7 @@
+ #include <infiniband/verbs_exp.h>
+ #endif
+ 
++#include "string.h"
```
Comment (bartoldeman) [inline easybuild/easyconfigs/u/UCX/UCX-1.11.2-missing_header.patch:23]: same here :)

**Review by bartoldeman — DISMISSED** [review]

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jsfc01.int.jusuf.sebastian.cluster - Linux rocky linux 8.4, x86_64, AMD EPYC 7742 64-Core Processor (zen2), 1 x NVIDIA GRID V100-4C, 460.73.01, Python 3.6.8
See https://gist.github.com/752880d6a3dea2a97a1238acad8407ef for a full test report.

**Review by Micket — APPROVED** [review]
LGTM

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/2b9ae658f559eb9c242e352c4c1b860c for a full test report.

**Comment by Micket** [comment]
Going in, thanks @sassy-crick!
